compiler/modules/types/listProj.bal

```diff
@@ -1,21 +1,19 @@
 // Untested full implementation of list projection.
 
 // Based on listMemberType
-public function listProj(Context cx, SemType t, int k) returns SemType {
-    if k < 0 {
-        return NEVER;
-    }
+public function listProj(Context cx, SemType t, SemType k) returns SemType {
     if t is UniformTypeBitSet {
         return (t & LIST) != 0 ? TOP : NEVER;
     }
     else {
-        return union(listProjBdd(cx, k, <Bdd>getComplexSubtypeData(t, UT_LIST_RO), (), ()),
-                     listProjBdd(cx, k, <Bdd>getComplexSubtypeData(t, UT_LIST_RW), (), ()));
+        IntSubtype|boolean keyData = intSubtype(k);
```

Comment:
If it's `false`, then the function will always return `never`, won't it?

So I would pass `IntSubtype|true` around.